src/greynirseq/nicenlp/models/multiclass.py

```diff
@@ -169,16 +170,45 @@ def decode(self, tokens):
         # Remove the leading space, see 'encode' comment.
         return super().decode(tokens)[1:]
 
-    def predict_labels(self, sentence):
-        tokens = self.encode(sentence)
-        word_mask = torch.tensor([self.word_start_dict[t] for t in tokens.tolist()])
-        word_mask[0] = 0
-        word_mask[-1] = 0
-        word_mask = word_mask.unsqueeze(0)
-        tokens = tokens.unsqueeze(0)
+    def predict_labels(self, sentences: List[str], batch_size=1) -> Iterable[List[str]]:
+        """Predicts NER labels of the given sentences.
+
+        Args:
+            sentences: An list of the rule-based tokenized sentences.
+            batch_size: The number of sentences to process in parallel.
+        Returns:
+            An iterable of the labels.
+        """
+        length = len(sentences)
+        for ndx in range(0, length, batch_size):
+            batch = sentences[ndx : min(ndx + batch_size, length)]
+            labels, pred_idx = self._predict_labels(batch)
+            yield from labels
```

Comment:
Does this call for a PR in GreynirSeqAPI ?

Reply:
Yes, since the default behavior is to return a Generator[Labels] instead of a single Labels.